includes/validation/class-amp-validation-manager.php

```diff
@@ -1893,38 +1890,105 @@ public static function validate_url( $url ) {
 		);
 	}
 
+	/**
+	 * Serialize validation error messages.
+	 *
+	 * In order to safely pass validation error messages through redirects with query parameters, they must be serialized
+	 * with a HMAC for security. The messages contain markup so the HMAC prevents tampering.
+	 *
+	 * @since 1.4.2
+	 * @see AMP_Validation_Manager::unserialize_validation_error_messages()
+	 *
+	 * @param string[] $messages Messages.
+	 * @return string Serialized.
+	 */
+	public static function serialize_validation_error_messages( $messages ) {
+		$encoded_messages = base64_encode( wp_json_encode( array_unique( $messages ) ) ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+		return wp_hash( $encoded_messages ) . ':' . $encoded_messages;
+	}
+
+	/**
+	 * Unserialize validation error messages.
+	 *
+	 * @since 1.4.2
+	 * @see AMP_Validation_Manager::serialize_validation_error_messages()
+	 *
+	 * @param string $serialized Serialized messages.
+	 * @return string[]|null
+	 */
+	public static function unserialize_validation_error_messages( $serialized ) {
+		$parts = explode( ':', $serialized, 2 );
+		if ( count( $parts ) !== 2 || wp_hash( $parts[1] ) !== $parts[0] ) {
+			return null;
+		}
+		return json_decode( base64_decode( $parts[1] ), true ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_decode
+	}
+
 	/**
 	 * Get error message for a validate URL failure.
 	 *
-	 * @param string $error_code Error code.
-	 * @return string Error message.
+	 * @param string $error_code    Error code.
+	 * @param string $error_message Error message, typically technical such as from HTTP status text or cURL error message.
+	 * @return string Error message with HTML markup.
 	 */
-	public static function get_validate_url_error_message( $error_code ) {
-		$check_error_log = __( 'Please check your server\'s PHP error logs; to do this you may need to enable WP_DEBUG_LOG.', 'amp' );
+	public static function get_validate_url_error_message( $error_code, $error_message = '' ) {
+		$check_error_log = sprintf(
+			/* translators: %s is link to Debugging in WordPress */
+			' ' . __( 'Please check your server PHP error logs; to do this you may need to <a href="%s" target="_blank">enable</a> <code>WP_DEBUG_LOG</code>.', 'amp' ),
+			esc_url( 'https://wordpress.org/support/article/debugging-in-wordpress/' )
+		);
+
+		if ( $error_message ) {
+			$error_message = ' ' . rtrim( $error_message, '.' ) . '.';
+		}
+
+		$support_forum_message = ' ' . sprintf(
+			/* translators: 1: Link to support forum. 2: Link to new topic form in support forum. */
```

Comment:
For consistency with other comments:
```suggestion
			/* translators: %1$s: Link to support forum. %2$s: Link to new topic form in support forum. */
```